the-collab-lab/tcl-27-smart-shopping-list
# Sk yh prohibit duplicate items

## Description

* When submitting a new item, we checked the database to see if there were any duplicates compared to what was about to be submitted
* Alerted user if they were about to submit a duplicate item and prohibited the item from being submitted to the database
* Used the toLowerCase() method and Regex to ignore case and punctuations when comparing items about to be submitted and items in the database
* Refactored changeHandler function to make it more readable

## Related Issue
Closes #6

## Acceptance Criteria
https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/6

6. As a user, I want to be alerted when I’m entering an item that is the same as something already on my list so I can avoid duplicate items.

AC:
* Show an error message if the user tries to submit a new item that has the exact same name as an existing item
* Show an error message if the user tries to submit a new item that has the same name as an existing item, where capitalization has been normalized and punctuation has been removed
* The user’s original input should be what gets saved in the database

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

From your terminal, pull from the main branch with `git pull origin main`. Pull this branch with `git pull origin sk-yh-prohibit-duplicate-items` and checkout branch with `git checkout sk-yh-prohibit-duplicate-items`

Add an item that is already included in the list and make sure that you are alerted that it is a duplicate item. Also try adding punctuation or casing the item to check that it disregards the punctuation/case and still alerts you that the item is a duplicate.
